bin/subst.ml

```diff
@@ -110,15 +110,18 @@ let subst_string s path ~map =
 ;;
 
 let subst_file path ~map =
-  let s = Io.read_file path in
-  let s =
-    if Path.is_root (Path.parent_exn path) && Package.is_opam_file path
-    then "version: \"%%" ^ "VERSION_NUM" ^ "%%\"\n" ^ s
-    else s
-  in
-  match subst_string s ~map path with
-  | None -> ()
-  | Some s -> Io.write_file path s
+  match Io.read_file path with
+  | exception Io.Large_file ->
+    User_warning.emit [ Pp.textf "Ignoring large file: %s" (Path.to_string path) ]
+  | s ->
+    let s =
+      if Path.is_root (Path.parent_exn path) && Package.is_opam_file path
```

Comment:
I know that this is from the original code but this looks like it needs to be adopted if we generate OPAM files in the `opam/` subdirectory.